packages/pyolite-kernel/src/worker.ts

```diff
@@ -17,11 +17,22 @@ async function loadPyodideAndPackages() {
   await pyodide.loadPackage(['matplotlib']);
   await pyodide.runPythonAsync(`
     import micropip
-    await micropip.install('${_pyoliteWheelUrl}')
+    # Doing it in two times is important, otherwise ipywidgets doesn't pick our mocks
```

Comment:
i kinda think we have to get these out of the critical path... loading `matplotlib` every time isn't fun, and this will start to get out of hand.

Reply:
Yeah, this doesn't scale. Let's remove it and install it manually in the Notebook example.

We might want to find a way for jupyterlite to have a plug-in system allowing users to define which packages/wheels they want to load automatically at the start. The same way ipython allows executing code when it starts.

Reply:
Right and in retrospect we don't ship `@jupyter-widgets/jupyterlab-manager` by default. So having `ipywidgets` only might be confusing with widgets not displaying if the prebuilt extension is not installed and served.

Reply:
yeah, and it's important folk are able to _own_ the versions of those... anything we hard-code in will be no fun. Now, we _could_ ship it as an optional "preferred" `federated_extensions` (either in the `jupyterlite` wheel, or fetched from the `jupyterlab_widgets-*.whl`), and ship a known-good tarball...

Reply:
also, regarding the imports: perhaps adopting a best-effort "runs-everywhere" pattern in the notebooks like:

```py
try:
    import pyolite
    await pyolite.patches.matplotlib()
except ImportError:
    pass
import matplotlib
```

with the _piece de resistance_ being if `importnb` works...